pkg/reconciler/instances/istio/reset/data/gatherer.go

```diff
@@ -84,3 +84,15 @@ func isPodReady(pod v1.Pod) bool {
 
 	return pod.ObjectMeta.DeletionTimestamp == nil
 }
+
+// FilterAnnotatedPods removes pods with annotation annotationKey from in podList
+func FilterAnnotatedPods(in v1.PodList, annotationKey string) (out v1.PodList) {
```

Comment:
The name is misleading. `FilterAnnotatedPods` could mean that you will retrieve pods with the given annotation whereas right now it is opposite.